server/ladders-remote.ts

```diff
@@ -149,4 +150,31 @@ export class LadderStore {
 	static async visualizeAll(username: string) {
 		return [`<tr><td><strong>Please use the official client at play.pokemonshowdown.com</strong></td></tr>`];
 	}
+
+	calculateElo(p1: User | null, p2: User | null, score: number, formatId: string) {
+		let K = 50;
+		const elo = p1?.mmrCache[formatId];
+		const foeElo = p2?.mmrCache[formatId];
+		if (!elo || !foeElo) return "unknown, please try again later";
+
+		// dynamic K-scaling (optional)
+		if (elo < 1200) {
```

Comment:
This formula doesn't seem accurate to the [server-side one](https://github.com/smogon/pokemon-showdown-client/blob/master/lib/ntbb-ladder.lib.php#L395).